### PR TITLE
docs: add Dialog header/footer sections

### PR DESCRIPTION
**Pull request by DiegoCardoso**
- Update Dialog documentation page to add sections for header and
footer.
- Refactor other examples to use the new API

### CODE REVIEW

File: articles/ds/components/dialog/index.asciidoc [hunk 1]
```diff
@@ -104,6 +167,11 @@ By default, the outer edges of a Dialog, as well as the space between its compon
 Any component contained within a Dialog can be marked and used as a drag handle by applying the `draggable` class name to it.
 You can choose whether or not to make the component's content draggable as well, or just the component itself.
 
+It is recommended to make non-modal Dialogs draggable so that the user can interact with content that might otherwise be obscured by the Dialog.
```
Comment (DiegoCardoso) [inline articles/ds/components/dialog/index.asciidoc:170]: There's one item here under discussion on the draft doc that was not added yet.

**Review by DiegoCardoso — COMMENTED** [review]
Keeping it as draft since we need the next alpha release.

File: articles/ds/components/dialog/index.asciidoc [hunk 2]
```diff
@@ -5,6 +5,8 @@ page-links:
   - 'Source: https://github.com/vaadin/web-components/tree/v{moduleNpmVersion:vaadin-dialog}/packages/vaadin-dialog[TypeScript] / https://github.com/vaadin/flow-components/tree/{moduleMavenVersion:com.vaadin:vaadin}/vaadin-dialog-flow-parent[Java]'
 ---
 
+:experimental:
+
 = Dialog
```
Comment (jouni) [inline articles/ds/components/dialog/index.asciidoc:10]: Never put any content/attributes/config before the main page heading. That messes up the placement of the TOC in the website. Might be worth a lint error…

```suggestion
= Dialog
:experimental:
```

File: articles/ds/components/dialog/index.asciidoc [hunk 3]
```diff
@@ -36,6 +38,72 @@ include::{root}/src/main/java/com/vaadin/demo/component/dialog/DialogBasic.java[
...
+
+[source,java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/dialog/DialogNoPadding.java[render,tags=snippet,indent=0,group=Java]
+----
+
+[source,css]
+----
+include::{root}/frontend/themes/docs/dialog.css[tags=dialog-no-padding,indent=0,group=Java]
+----
```
Comment (jouni) [inline articles/ds/components/dialog/index.asciidoc:104]: This file is unnecessary now, isn't it?

**Review by jouni — COMMENTED** [review]
I’m unsure if we want to capitalize all occurrences of "Dialog" and "Dialogs". Seems a bit excessive and distracting to me. I’d try to limit that to places where we explicitly say "the Dialog component".

File: articles/ds/components/dialog/index.asciidoc (hunk 3 above)
Comment (DiegoCardoso) [reply articles/ds/components/dialog/index.asciidoc]: Good catch! Removed.

File: articles/ds/components/dialog/index.asciidoc [hunk 4]
```diff
@@ -104,6 +161,11 @@ By default, the outer edges of a Dialog, as well as the space between its compon
 Any component contained within a Dialog can be marked and used as a drag handle by applying the `draggable` class name to it.
 You can choose whether or not to make the component's content draggable as well, or just the component itself.
 
+It is recommended to make non-modal Dialogs draggable so that the user can interact with content that might otherwise be obscured by the Dialog.
+For example, a Dialog for taking notes, or adding widgets to a dashboard using drag and drop, can offer a better experience by allowing the user to move the Dialog around.
+
+Modal Dialogs don't benefit from being draggable as their modality curtain (the dark overlay behind the dialog) obscures the underlying user interface.
```
Comment (dougjewsbury) [inline articles/ds/components/dialog/index.asciidoc:167]:
```suggestion
Modal Dialogs do not benefit from being draggable, as their modality curtain (the dark overlay behind the dialog) obscures the underlying user interface.
```

File: articles/ds/components/dialog/index.asciidoc [hunk 5]
```diff
@@ -36,6 +37,67 @@ include::{root}/src/main/java/com/vaadin/demo/component/dialog/DialogBasic.java[
...
+----
+
+[source,java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/dialog/DialogHeader.java[render,tags=*,indent=0,group=Java]
+----
+--
+
+=== Footer
+Buttons for closure actions such as _Save_, _Cancel_, _Delete_ etc should be placed in the footer.
```
Comment (dougjewsbury) [inline articles/ds/components/dialog/index.asciidoc:63]:
```suggestion
Buttons for closure actions, such as _Save_, _Cancel_, _Delete_, etc., should be placed in the footer. 
```

File: articles/ds/components/dialog/index.asciidoc (hunk 1 above)
Comment (rolfsmeds) [reply articles/ds/components/dialog/index.asciidoc:164]: Which is that?

**Review by rolfsmeds — APPROVED** [review]
LGTM. I presume the parts about server-side modality were excluded since they don't have anything to do with header + footer?

File: articles/ds/components/dialog/index.asciidoc (hunk 1 above)
Comment (DiegoCardoso) [reply articles/ds/components/dialog/index.asciidoc:164]: It's not exactly here, but before the paragraph which starts with "Any component...". to add this:

> The default areas from which a Dialog can be dragged depend on whether the built-in header is used:
> - If the built-in header or footer is used, they function as the Dialog’s default drag handles
> - Without the built-in header, any empty space within the dialog functions as a drag handle

**Comment by DiegoCardoso** [comment]
> LGTM. I presume the parts about server-side modality were excluded since they don't have anything to do with header + footer?

Yeah, I didn't add it because I wasn't sure if it should be added here or later. But I can add it, if it's ok.

**Comment by rolfsmeds** [comment]
> > LGTM. I presume the parts about server-side modality were excluded since they don't have anything to do with header + footer?
> 
> Yeah, I didn't add it because I wasn't sure if it should be added here or later. But I can add it, if it's ok.

No strong opinion either way :)

File: articles/ds/components/dialog/index.asciidoc (hunk 1 above)
Comment (rolfsmeds) [reply articles/ds/components/dialog/index.asciidoc:164]: I think it's good enough as it is now. But I think I would move the examples down a bit, just before the paragraph starting with "It is recommended to make non-modal Dialogs draggable..."

Comment (DiegoCardoso) [reply articles/ds/components/dialog/index.asciidoc]: Done.

File: frontend/demo/component/dialog/dialog-draggable.ts [hunk 6]
```diff
@@ -27,10 +26,21 @@ export class Example extends LitElement {
 
   render() {
     return html`
-      <!-- tag::snippet[] -->
+      <!-- tag::snippet1[] -->
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-draggable.ts:29]: The build fails with an error, please fix:

```
warn /Users/serhii/vaadin/docs/articles/ds/components/dialog/index.asciidoc, line 3 - detected unclosed tag 'snippet1' starting at line 29 of include file: /Users/serhii/vaadin/docs/frontend/demo/component/dialog/dialog-draggable.ts
```

**Review by web-padawan — CHANGES_REQUESTED** [review]

File: frontend/demo/component/dialog/dialog-draggable.ts (hunk 6 above)
Comment (DiegoCardoso) [reply frontend/demo/component/dialog/dialog-draggable.ts]: Done.

File: frontend/demo/component/dialog/dialog-footer.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        })}"
+      ></vaadin-dialog>
+      <!-- end::snippet[]  -->
+      <vaadin-button @click="${() => (this.dialogOpened = true)}"> Show dialog </vaadin-button>
+    `;
+  }
+
+  dialogRenderer = (root: HTMLElement) => {
+    render(html`<p>Are you sure you want to delete this user permanently?</p>`, root);
+  };
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-footer.ts:69]: This seems to be unused - as the `.renderer` is set to inline arrow function. Should it be removed?

File: frontend/demo/component/dialog/dialog-no-padding.ts [hunk 8]
```diff
@@ -5,15 +5,16 @@ import { customElement, state } from 'lit/decorators.js';
...
 import '@vaadin/dialog';
-import '@vaadin/horizontal-layout';
-import '@vaadin/scroller';
-import '@vaadin/text-area';
-import '@vaadin/text-field';
-import '@vaadin/vertical-layout';
+
+import '@vaadin/grid';
+import '@vaadin/grid/vaadin-grid-selection-column.js';
+import '@vaadin/button';
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-no-padding.ts:12]: There is an import for '@vaadin/button' above, so this one can be removed.

File: frontend/demo/component/dialog/dialog-footer.ts (hunk 7 above)
Comment (DiegoCardoso) [reply frontend/demo/component/dialog/dialog-footer.ts]: Good catch. Removed.

File: frontend/demo/component/dialog/dialog-header.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        .renderer="${guard(this.user, () => this.dialogRenderer)}"
+      ></vaadin-dialog>
+      <!-- end::snippet[]  -->
+      <vaadin-button @click="${() => (this.dialogOpened = true)}"> Show dialog </vaadin-button>
+    `;
+  }
+
+  dialogRenderer = (root: HTMLElement) => {
+    console.log('got here', this.user, this.dialogOpened);
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply frontend/demo/component/dialog/dialog-header.ts]: 🤦🏻‍♂️  removed.

File: frontend/demo/component/dialog/dialog-no-padding.ts (hunk 8 above)
Comment (DiegoCardoso) [reply frontend/demo/component/dialog/dialog-no-padding.ts]: Removed.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**